exp/query/api.go

```diff
@@ -315,30 +319,108 @@ func (a *API) vectorSearch(
 	return objs, distance, nil
 }
 
-func (a *API) loadOrDownloadLSM(ctx context.Context, collection, tenant string) (*lsmkv.Store, string, error) {
-	localPath := path.Join(a.offload.DataPath, strings.ToLower(collection), strings.ToLower(tenant), defaultLSMRoot)
+type LSMEnsurer interface {
```

Comment:
I would remove this interface :). 

Rationale are
1. This is used only inside this query package for now. No external package consumes it. Even any consumer wants to mock it, they can create this interface on the consumer side not on the implementation side.
2. Go interfaces are designed for [consumer side](https://go.dev/wiki/CodeReviewComments#interfaces)

To give concrete example
1. We have [SchemaQuerier](https://github.com/weaviate/weaviate/blob/main/exp/query/api.go#L77-L80) interface on `query` package [where I consume it.](https://github.com/weaviate/weaviate/blob/main/exp/query/api.go#L67)
2. But it's implemented in another package `queryschema` here. Why?
4. Because, now `query` and `queryschema` package is **not coupled**. And I link them [only](https://github.com/weaviate/weaviate/blob/main/cmd/weaviate/main.go#L62) in the [main](https://github.com/weaviate/weaviate/blob/main/cmd/weaviate/main.go#L79). If we both define and implement `SchemaQuerier` interface in `queryschema` package then we would have to use that interface in `query` package as `queryschema.SchemaQuerier` that creates coupling between `query` and `queryschema`.

In other words, it's **almost always wrong** to have both definition and implementation of interface in **same** package :). Few exceptions of course like in stdlib.

Reply:
Thanks, I got rid of the interface. I think at some point it would be nice to see if we can break out the download functionality from the API, but for now I'm just using the API as is